app/src/main/java/de/tum/in/tumcampusapp/component/tumui/feedback/FeedbackController.java

```diff
@@ -0,0 +1,332 @@
+    }
+
+    String getCurrentPhotoPath() {
+        return mCurrentPhotoPath;
+    }
+
+    void startTakingPicture(Activity activity) {
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        // Ensure that there's a camera activity to handle the intent
+        if (takePictureIntent.resolveActivity(mContext.getPackageManager()) != null) {
+            // Create the File where the photo should go
+            File photoFile = null;
+            try {
+                photoFile = ImageUtils.createImageFile(mContext);
+                // Save a file: path for use with ACTION_VIEW intents
+                mCurrentPhotoPath = photoFile.getAbsolutePath();
+            } catch (IOException e) {
+                Utils.log(e);
+            }
+            // Continue only if the File was successfully created
+            if (photoFile != null) {
+                Utils.log(mContext.getExternalFilesDir(Environment.DIRECTORY_PICTURES).getAbsolutePath());
+                Uri photoURI = FileProvider.getUriForFile(mContext,
+                        "de.tum.in.tumcampusapp.fileprovider",
+                        photoFile);
+                takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, photoURI);
+                activity.startActivityForResult(takePictureIntent, REQUEST_TAKE_PHOTO);
+            }
+        }
+    }
+
+    void openGallery(Activity activity) {
+        Intent intent = new Intent();
+        intent.setType("image/*");
+        intent.setAction(Intent.ACTION_GET_CONTENT);//
+        activity.startActivityForResult(Intent.createChooser(intent, "Select File"), REQUEST_GALLERY);
+    }
+
+    private boolean isEmailValid(String email) {
+        return Patterns.EMAIL_ADDRESS.matcher(email).matches();
+    }
+
+    void sendFeedback(Activity activity, Feedback feedback, String lrzId) {
+        // adjust information that is to be sent to the server
+        if (!feedback.getIncludeEmail()) {
+            feedback.setEmail("");
+        }
+        if (!feedback.getIncludeLocation()) {
+            feedback.setLongitude(0);
+            feedback.setLatitude(0);
+        }
+        feedback.setImageCount(feedback.getPicturePaths().size());
+
+        Utils.log("Feedback: " + feedback.toString());
+
+        imagesSent = 0;
+        stopListeningForLocation();
+
+        if (feedback.getIncludeEmail() && !isEmailValid(feedback.getEmail())) {
+            Toast.makeText(activity.getApplicationContext(), "Email is not valid", Toast.LENGTH_SHORT).show();
+            return;
+        }
+
+        showProgressBarDialog();
+        TUMCabeClient client = TUMCabeClient.getInstance(mContext);
+        client.sendFeedback(feedback, new Callback<Success>() {
+            @Override
+            public void onResponse(@NonNull Call<Success> call, @NonNull Response<Success> response) {
+                Success success = response.body();
+                if (success != null && success.wasSuccessfullySent()) {
```

Comment:
Please keep the error handling consistent, I like how you did it in `sendFeedbackImages()`, but two different ways to handle it is confusing